javalib/src/main/scala/java/util/concurrent/ThreadLocalRandom.scala

```diff
@@ -105,8 +105,9 @@ class ThreadLocalRandom extends Random {
   def nextDouble(least: Double, bound: Double): Double = {
     if (least >= bound)
       throw new IllegalArgumentException()
-
-    nextDouble() * (bound - least) + least
+    // Ensure that rangeSize != Double.PositiveInfinity - Issue #2144
+    val rangeSize =  Math.min(bound - least, Double.MaxValue)
+    nextDouble() * rangeSize + least
```

Comment:
That does not seem correct to me. For `least = MinValue, bound = MaxValue`, this would produce only values betwen `MinValue` and `0`, but not between `0` and `MaxValue`.